### PR TITLE
Postfacto deployment on Tanzu Kubernetes Grid (vSphere)

### PR DESCRIPTION
**Pull request by th3n3rd**
* A short explanation of the proposed change:
Provided scripts to deploy and upgrade Postfacto on Tanzu Kubernetes Grid. Documented the deployment and upgrade steps.

* An explanation of the use cases your change solves
Allows Alex the admin to install Postfacto on Tanzu Kubernetes Grid.

* Links to any other associated PRs

* [x] I have reviewed the [contributing guide](https://github.com/pivotal/postfacto/blob/master/CONTRIBUTING.md)

* [x] I have made this pull request to the `master` branch

* [x] I have run all the tests using `./test.sh`.

* [x] I have added the [copyright headers](https://github.com/pivotal/postfacto/blob/master/license-header.txt) to each new file added

* [x] I have given myself credit in the [humans.txt](https://github.com/pivotal/postfacto/blob/master/humans.txt) file (assuming I want to)

### CODE REVIEW

File: deployment/deploy-tkg.sh [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+#
+#
+# You should have received a copy of the GNU Affero General Public License
+#
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+set -euo pipefail
+
+if [ $# -lt 1 ]; then
+  echo "usage: ./deploy.sh <app-name> [<kubeconfig-path>]"
```
Comment (textbook) [inline deployment/deploy-tkg.sh:35]: Do we actually use the `kubeconfig-path` anywhere?

File: deployment/upgrade-tkg.sh [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+#
+#
+# You should have received a copy of the GNU Affero General Public License
+#
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+set -euo pipefail
+
+if [ $# -lt 1 ]; then
+  echo "usage: ./upgrade.sh <app-name> [<kubeconfig-path>]"
```
Comment (textbook) [inline deployment/upgrade-tkg.sh:35]: Same

File: deployment/README.md [hunk 3]
```diff
@@ -4,6 +4,39 @@
 
 So you're ready to set Postfacto up, choose a name for your app. We'll refer to this as `app-name` from now on.
 
+## Tanzu Kubernetes Grid (vSphere)
+
+### Prerequisites
+
+1. A vSphere 6.7u3 instance with Tanzu Kubernetes Grid (TKG) provisioned
```
Comment (textbook) [inline deployment/README.md:11]: Is that vSphere 6.7u3 _and up_?

File: .travis.yml [hunk 4]
```diff
@@ -33,24 +33,10 @@ jobs:
...
-    deploy:
-      - provider: releases
-        draft: true
-        skip_cleanup: true
-        overwrite: true
-        on:
-          tags: true
-        api_key:
-          secure: dPDprfJo3c5iCk5fcPR7tgRMz5OcCF6IAPRx1/WMAq1FJjgdkD6iu628/AAgezN/rlTDiqZpfx3nj3UhTSUOeBZgEaRbpvpIc3bLc2RABqqxQNuL2g+DNt3L7XYcq8PLHQBVIHmRKp8DK1XJCaG3LUEuYZD9gKaiftMwbvgsgfVOFasslyypgDTi/NdPm1tNsDdKvYVe9dkZpDpHClJsoIQLl/gtNvrJREFTF2w0T8oSdccsoQekAAC4kXWiL5lgXIKHDpQTo1K8oXrl/hUxuw0e1fGKA6R0kNZliA6cQYG3OOb0a9JhplzO3jGWvlZdmJluWgy0yUn+jRZZOXiQVJOZCd2Lht3JN8p6x9D91S43forMV7ZmrW3XPuEocGIwVdY7yVKEsHAiAK3Ih3VbMuUn1ITvKB8F4idyurWEL9cQwsszX9pFSOriAH5mhIQs7vwp6uvFJIoADGiS//wEqcAwzsFWHCDLBPYRTk9Zp17MSgUj8+PDBo29pbbWtx6+14w82LVQQdnyoReDiZR5G1ItxSx6xjWIeGlyo8yu4xmNq9nufPoj6ZkGDbAJtH2YeaGIEcltjXzIalmBdmoVp+6SKyr8EFnolgWd+ns8KkCvLqxp3fwuz4OD+B619Au1/8WUrHRdetChlvxzEHVx3wnwWnw0zbCBh/cq0mWUdzs=
-        file_glob: true
```
Comment (textbook) [inline .travis.yml:48]: I suspect we still need this to allow `postfacto-*.tgz`

File: deployment/upgrade-tkg.sh (hunk 2 above)
Comment (th3n3rd) [reply deployment/upgrade-tkg.sh]: Done.

File: deployment/deploy-tkg.sh (hunk 1 above)
Comment (th3n3rd) [reply deployment/deploy-tkg.sh]: Removed as unused. We ended up remove it's usage to simplify the script.

File: deployment/README.md (hunk 3 above)
Comment (th3n3rd) [reply deployment/README.md]: Updated.

File: .travis.yml (hunk 4 above)
Comment (th3n3rd) [reply .travis.yml:48]: I've added the property to the deploy action of "Build & Test Package". Is there a way to test it works properly with a draft release?

Comment (textbook) [reply .travis.yml:48]: I don't think there's a way to test it really. We know it worked from the "Build & Deploy Helm" step, and it matches the description in https://docs.travis-ci.com/user/deployment/releases/#uploading-multiple-files

Comment (j-lea) [reply .travis.yml:48]: We have tested it by pushing to my fork of Postfacto, and that managed to upload https://github.com/j-lea/postfacto/releases/tag/j8

Comment (th3n3rd) [reply .travis.yml:48]: I am going to resolve this thread, reopen if you need more info on this.

**Pull request merged**